NPE when opening cache history or stored lists

NPE when opening cache history or stored lists

Support Ticket: 583918
cgeo 2024.03.11-NB-e0a0232

```
03-11 18:23:20.827 14277 14277 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{cgeo.geocaching/cgeo.geocaching.CacheListActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'int cgeo.geocaching.SearchResult.getTotalCount()' on a null object reference
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4011)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4161)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:114)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:144)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:101)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2627)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:257)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:368)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8825)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:572)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1049)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int cgeo.geocaching.SearchResult.getTotalCount()' on a null object reference
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at cgeo.geocaching.CacheListActivity.resultIsOfflineAndLimited(CacheListActivity.java:2029)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at cgeo.geocaching.CacheListActivity.refreshActionBarTitle(CacheListActivity.java:530)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at cgeo.geocaching.CacheListActivity.updateTitle(CacheListActivity.java:337)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at cgeo.geocaching.CacheListActivity.onCreateLoader(CacheListActivity.java:1941)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at androidx.loader.app.LoaderManagerImpl.createAndInstallLoader(LoaderManagerImpl.java:383)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at androidx.loader.app.LoaderManagerImpl.restartLoader(LoaderManagerImpl.java:447)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at cgeo.geocaching.CacheListActivity.restartCacheLoader(CacheListActivity.java:1473)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at cgeo.geocaching.CacheListActivity.onCreate(CacheListActivity.java:488)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8757)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8729)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1475)
03-11 18:23:20.827 14277 14277 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3985)
```

--- a/main/src/main/java/cgeo/geocaching/CacheListActivity.java
+++ b/main/src/main/java/cgeo/geocaching/CacheListActivity.java
@@ -2026,7 +2026,7 @@ public class CacheListActivity extends AbstractListActivity implements FilteredA
     }
 
     private boolean resultIsOfflineAndLimited() {
-        return type.isStoredInDatabase && offlineListLoadLimit > 0 && search.getTotalCount() > offlineListLoadLimit && search.getCount() == offlineListLoadLimit;
+        return type.isStoredInDatabase && offlineListLoadLimit > 0 && search != null && search.getTotalCount() > offlineListLoadLimit && search.getCount() == offlineListLoadLimit;
     }
 
     /**

